File: Easy/2025-10-29-3370-Smallest-Number-With-All-Set-Bits/solution.java
class Solution {
    public int smallestGoodBase(int n) {
        for (int m = 60; m >= 1; m--) {
            long k = (long) Math.pow(2, m) - 1;
            if (k >= n) {
                long base = (long) Math.pow(n, 1.0 / m);
                while (true) {
                    long sum = 0, current = 1;
                    for (int i = 0; i <= m; i++) {
                        sum += current;
                        current *= base;
                    }
                    if (sum >= n) {
                        return (int) sum;
                    }
                    base++;
                }
            }
        }
        return n;
    }
}

File: Easy/2025-10-29-3370-Smallest-Number-With-All-Set-Bits/solution.js
var smallestGoodBase = function(n) {
    let maxPower = Math.floor(Math.log(n) / Math.log(2));
    for (let k = maxPower; k >= 1; k--) {
        let base = Math.floor(Math.pow(n, 1 / k));
        let sum = (base ** (k + 1) - 1) / (base - 1);
        if (sum === n) {
            return base.toString(2).repeat(k + 1);
        }
    }
    return (n).toString(2).repeat(1);
};